libs/sr/ui/src/Relic/RelicInventory.tsx

```diff
@@ -20,86 +10,89 @@ const numToShowMap = { xs: 10, sm: 12, md: 24, lg: 24, xl: 24 }
 
 export function RelicInventory() {
   const { database } = useDatabaseContext()
+  const [total, setTotal] = useState(database.relics.values.length)
 
-  const { relicsIds, totalRelicsNum } = useMemo(() => {
-    const relicsIds = database.relics.keys
-    const totalRelicsNum = relicsIds.length
-    return { relicsIds, totalRelicsNum }
-  }, [database])
+  useEffect(
+    () =>
+      database.relics.followAny(
+        (_k, r) =>
+          ['new', 'remove'.includes(r)] &&
+          setTotal(database.relics.values.length)
+      ),
+    [database]
+  )
+
+  const relicIds = useMemo(() => {
+    const relics = database.relics.values
+    const relicIds = relics.map((r) => r.id)
+    return relicIds
+  }, [database.relics.values])
```

Comment:
Using `[database.relics.values]` as the dependency array here would actually trigger a recalc every time, since it would create a new array every time its called. 
```suggestion
  }, [total])
```